java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -211,4 +216,65 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
       addMetadata(pwsResult);
     }
   }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param dedupSiteUrls whether or not to dedup PwPairs by result urls.
+   * @param minRank the minimum rank to keep in the list.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(boolean dedupSiteUrls, double minRank) {
+    // Get all valid PwPairs.
+    List<PwPair> allPwPairs = new ArrayList<>();
```

Comment:
You mean the siteUrls hashset?  It wouldn't be valid if we constructed it here because there should be no entry before we've found the highest ranked pair.